SwiftUI is now out of Beta 🎉 (#128)
wwt/SwiftCurrent: README.md

SwiftCurrent is a library that lets you easily manage journeys through your Swift application.

> It comes with built-in support for UIKit and SwiftUI app-routing. In SwiftCurrent workflows are a sequence of operations. Those operations are normally showing views in an application. The workflow describes the sequence of views and manages what view should come next. Your views are responsible for performing necessary tasks before proceeding forward in the workflow, like processing user input.

Reviewer: More Grammarbot

```suggestion
It comes with built-in support for UIKit and SwiftUI app-routing. In SwiftCurrent workflows are a sequence of operations. Those operations usually show views in an application. The workflow describes the sequence of views and manages what view should come next. Your views are responsible for performing necessary tasks before proceeding forward in the workflow, like processing user input.
```

- **A Developer Friendly API**: The library was built with developers in mind. It started with a group of developers talking about the code experience they desired. Then the library team took on whatever complexities were necessary to bring them that experience.
- **Compile-time safety**: We tell you at compile time everything we can so you know things will work.
- **Minimal Boilerplate**: We have hidden this as much as possible. We hate it as much as you do and are constantly working on cutting the cruft.

#### From there, we created a library that:
- **Isolates your views**:  You can design your views so that they are unaware of the view that will come next.
- **Easily reorders views**: Changing view order is as easy as ⌘+⌥+\[ (moving the line up or down)
- **Composes workflows together**: Create branching flows easily by joining workflows together.
- **Creates conditional flows**: Make your flows robust and handle ever-changing designs. Need a screen only to show up sometimes? Need a flow for person A and another for person B? We've got you covered.

# Quick Start

> Why show a quick start when we have an example app? Because it's so easy to get started, we can just drop in 2 code snippets and you're ready to go! This quick start uses Swift Package Manager and SwiftUI, but for other approaches, [see our installation instructions](https://wwt.github.io/SwiftCurrent/installation.html).

Reviewer: ```suggestion
Why show a quick start when we have an example app? Because it's so easy to get started, we can drop in 2 code snippets, and you're ready to go! This quickstart uses Swift Package Manager and SwiftUI, but for other approaches, [see our installation instructions](https://wwt.github.io/SwiftCurrent/installation.html).
```

.package(url: "https://github.com/wwt/SwiftCurrent.git", .upToNextMajor(from: "4.5.0")),
...
.product(name: "SwiftCurrent", package: "SwiftCurrent"),
.product(name: "SwiftCurrent_SwiftUI", package: "SwiftCurrent")
```
Then make your first FlowRepresentable view:
```swift
import SwiftCurrent
import SwiftUI
struct OptionalView: View, FlowRepresentable {
    weak var _workflowPointer: AnyFlowRepresentable?
    let input: String
    init(with args: String) { input = args }
    var body: some View { Text("Only shows up if no input") }
    func shouldLoad() -> Bool { input.isEmpty }
}
struct ExampleView: View, PassthroughFlowRepresentable {
    weak var _workflowPointer: AnyFlowRepresentable?
    var body: some View { Text("This is ExampleView!") }
}
```
Then from your `ContentView` or whatever view (or app) you'd like to contain the workflow, add the following view to the body: 
```swift
import SwiftCurrent_SwiftUI
// ...
var body: some View { 
    // ... other view code (if any)
    WorkflowLauncher(isLaunched: .constant(true), startingArgs: "Skip optional screen") {
        thenProceed(with: OptionalView.self) {
            thenProceed(with: ExampleView.self)
        }
    }
}
```

> And just like that you've got a workflow! You can now add more items to it, or re-order the items that are there. To understand more of how this works and what is there [check out our developer docs.](https://wwt.github.io/SwiftCurrent/How%20to%20use%20SwiftCurrent%20with%20SwiftUI.html)

Reviewer: ```suggestion
And just like that, you've got a workflow! You can now add more items to it or re-order the items that are there. To understand more of how this works and what is there [check out our developer docs.](https://wwt.github.io/SwiftCurrent/How%20to%20use%20SwiftCurrent%20with%20SwiftUI.html)
```

# Check out our example apps
We have [example apps](https://github.com/wwt/SwiftCurrent/tree/main/ExampleApps) for both SwiftUI and UIKit that show SwiftCurrent in action. They're even tested so you can see what it's like to test SwiftCurrent code. To run it locally, start by cloning the repo, open `SwiftCurrent.xcworkspace` and then run the `SwiftUIExample` scheme or the `UIKitExample` scheme. 

> # [See this to learn more](https://wwt.github.io/SwiftCurrent/index.html)

Reviewer: This linking back to the README feels odd - link to a more specific page or update the text?

```suggestion
# [Learn more in our Docs](https://wwt.github.io/SwiftCurrent/index.html)
```
